diffmpm/shapefn.py

```diff
@@ -1,16 +1,108 @@
 import jax.numpy as jnp
+from jax import jacobian, vmap, lax
 
-def Linear1DShapefn(xi):
-    """
-    Linear 1D shape function in natural coordinates
+
+class ShapeFn:
```

Comment:
There is also GIMP shape functions even for 1D...but let's go ahead with this. We can think about GIMP shape functions later

Reply:
Ah okay. That should be easy to add via a kwarg so shouldn't be very difficult since gradient calculation is automatic so we'll just have to specify the function and it's transform.